torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2394,6 +2393,11 @@ def _recording(self):
         finally:
             self.is_recording = False
 
+    # Unlike set_replacement, this records a shapeenv event
+    @record_shapeenv_event()
+    def _rename_unbacked_to(self, orig_s: sympy.Expr, new_s: sympy.Expr):
```

Comment:
both should be symbols, not Expr. Can you assert that they are also unbacked?

Reply:
I'm going to add this assert in https://github.com/pytorch/pytorch/pull/124782 as I changed some details about how the call site works there

Reply:
https://github.com/pytorch/pytorch/pull/124782/files#diff-03557db7303b8540f095b4f0d9cd2280e1f42f534f67d8695f756ec6c02d3ec7